### PR TITLE
[Feature][CI] Add Dead Link Checker 

### LINKED ISSUE #9999: [Feature][CI] Add Dead Link Checker
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Add Dead Link Checker

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wuchunfu**
## Purpose of the pull request
close #9999

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Review by davidzollo — APPROVED** [review]
LGTM

**Comment by dailidong** [comment]
Please handle the dead link pngs

**Review by davidzollo — DISMISSED** [review]
Please handle the dead link pngs

**Comment by wuchunfu** [comment]
> Please handle the dead link pngs

I don't know what those two pngs are, can anyone help me?

**Comment by zhongjiajie** [comment]
But we already have the dead link checker in our docs.yaml https://github.com/apache/dolphinscheduler/blob/dev/.github/workflows/docs.yml#L48-L60 what the difference is? and could we combine them?

**Comment by wuchunfu** [comment]
> But we already have the dead link checker in our docs.yaml https://github.com/apache/dolphinscheduler/blob/dev/.github/workflows/docs.yml#L48-L60 what the difference is? and could we combine them?

Well, close it if you don't need my contribution this time.

File: .github/workflows/dead-link-checker.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Dead Link Checker
+
+on:
+  pull_request:
+  schedule:
+    - cron: '0 18 * * *'  # TimeZone: UTC 0
```
Comment (zhongjiajie) [inline .github/workflows/dead-link-checker.yaml]: Hi @wuchunfu , maybe we should add this setting or others setting we miss into `docs.yaml` CI configuration. WDYT

Comment (wuchunfu) [reply .github/workflows/dead-link-checker.yaml]: Thanks jiajie for the review, I'll finish it

**Review by zhongjiajie — COMMENTED** [review]
LGTM, thanks~

**Review by zhongjiajie — APPROVED** [review]

**Comment by zhongjiajie** [comment]
I rerun the failed CI

**Comment by zhongjiajie** [comment]
Restarted it again

**Review by kezhenxu94 — APPROVED** [review]

**Comment by devosend** [comment]
This pr should be cherry-pick to milestone 3.0.0-beta-2? @zhongjiajie @wuchunfu

**Comment by kezhenxu94** [comment]
> This pr should be cherry-pick to milestone 3.0.0-beta-2? @zhongjiajie @wuchunfu

This is just CI settings, not functionality related, I think we don't need to back port this to previous version.

**Comment by devosend** [comment]
> > This pr should be cherry-pick to milestone 3.0.0-beta-2? @zhongjiajie @wuchunfu
> 
> This is just CI settings, not functionality related, I think we don't need to back port this to previous version.

But is will affect cherry-pick of #10325

**Comment by kezhenxu94** [comment]
> > > This pr should be cherry-pick to milestone 3.0.0-beta-2? @zhongjiajie @wuchunfu
> > 
> > 
> > This is just CI settings, not functionality related, I think we don't need to back port this to previous version.
> 
> But is will affect cherry-pick of #10325

OK if that's the case, you can cherry pick this. It's not functionality related so you can choose to or not to cherry pick it, if other depends on this you can do it

**Comment by zhongjiajie** [comment]
> > > > This pr should be cherry-pick to milestone 3.0.0-beta-2? @zhongjiajie @wuchunfu
> > > 
> > > 
> > > This is just CI settings, not functionality related, I think we don't need to back port this to previous version.
> > 
> > 
> > But is will affect cherry-pick of #10325
> 
> OK if that's the case, you can cherry pick this. It's not functionality related so you can choose to or not to cherry pick it, if other depends on this you can do it

Agrees with that we can or can not cherry pick this to nest release branch

**Pull request merged**